File: DayMap V2/mac/DayMap/ProjectTableRowView.m
//
//  ProjectTableRowView.m
//  DayMap
//
//  Created by Chad Seldomridge on 8/19/13.
//  Copyright (c) 2013 Monk Software LLC. All rights reserved.
//

#import "ProjectTableRowView.h"
#import "NSColor+WS.h"

@implementation ProjectTableRowView

- (BOOL)isOpaque {
	return YES;
}

- (void)drawBackgroundInRect:(NSRect)dirtyRect {
	[[NSColor dayMapBackgroundColor] set];
	NSRectFill(dirtyRect);
}

@end

File: DayMap/ProjectTableRowView.swift
//
//  ProjectTableRowView.swift
//  DayMap
//
//  Created by Chad Seldomridge on 11/24/17.
//  Copyright © 2017 Whetstone Apps. All rights reserved.
//

import Cocoa

class ProjectTableRowView: NSTableRowView {
	
	override var isOpaque: Bool {
		return true
	}
	
	override func drawSelection(in dirtyRect: NSRect) {
		if isSelected && selectionHighlightStyle == .regular {
			if isEmphasized {
				NSColor.selectedControlColor.set()
			}
			else {
				NSColor.secondarySelectedControlColor.set()
			}
			__NSRectFillUsingOperation(dirtyRect, .copy)
		}
		else {
			super.drawSelection(in: dirtyRect)
		}
	}
}
